cmd/mock/api-server/rest/app/root.go

```diff
@@ -54,11 +61,30 @@ func Command() *cli.RootCommand {
 
 				r := resolver.NewComponentResolver(gitResolver, templateStore, secretStore)
 				backend := mockbackend.NewMockBackend(r)
-				rest.RunNewRestServer(backend, r, port, apiAuthKey)
+				options := rest.NewServerOptions()
+				// apply auth options based on input
+				applyAuthenticationOptions(options, apiAuthKey, tokenAuthFile)
+				rest.RunNewRestServer(backend, r, port, options)
 				return nil
 			},
 		},
 	}
 
 	return command
 }
+
+func applyAuthenticationOptions(options *rest.ServerOptions, apiAuthKey string, tokenAuthFile string) {
```

Comment:
same as above, why not just have this create and return a `*rest.ServerOptions`?